src/combinatorics/core/elements/ArrayElement.java

```diff
@@ -0,0 +1,76 @@
+package combinatorics.core.elements;
+
+import java.util.ArrayList;
+
+import combinatorics.core.Comparer;
+
+public class ArrayElement implements Element {
+	
+	public Element[] value;
+	
+	public ArrayElement(Element[] value) {
+		this.value = value;
+	}
+	
+	public ArrayElement(ArrayList<?> value) {	//This constructor exists for the cases when it's easier to work with lists than arrays during a construction (eg appending when you don't know the size).
+		this.value = new Element[value.size()];
+		for (int i = 0; i < value.size(); i++) {
+			this.value[i] = (Element) value.get(i);
+		}
+	}
+
+	public Element add(Element other) {
+		if (other instanceof ArrayElement) {	//n-dimensional component-wise addition
+			if (this.value.length == ( (ArrayElement) other).value.length) {	//makes sure the arrays are the same size
+				for (int i = 0; i < this.value.length; i++) {
+					this.value[i] = this.value[i].add( ( (ArrayElement) other ).value[i]);
+				}
+			}
+		}
+		else {
+			for (int i = 0; i < this.value.length; i++) {	//add other value to each component of array
+				this.value[i] = this.value[i].add(other);
+			}
+		}
+		return this;
+	}
```

Comment:
You could overload `add` instead of using the `instanceof` operator, like this:

```suggestion
	public Element add(ArrayElement other) {
//n-dimensional component-wise addition
		if (this.value.length == ( (ArrayElement) other).value.length) {	//makes sure the arrays are the same size
			for (int i = 0; i < this.value.length; i++) {
				this.value[i] = this.value[i].add( ( (ArrayElement) other ).value[i]);
			}
		}
		return this;
	}
	public Element add(ArrayElement other) {
		for (int i = 0; i < this.value.length; i++) {	//add other value to each component of array
			this.value[i] = this.value[i].add(other);
		}
		return this;
	}
```